debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -357,25 +356,33 @@ public Scn getCurrentScn() throws SQLException {
         });
     }
 
-    public OracleConnection executeLegacy(String... sqlStatements) throws SQLException {
-        return executeLegacy(statement -> {
-            for (String sqlStatement : sqlStatements) {
-                if (sqlStatement != null) {
-                    statement.execute(sqlStatement);
-                }
+    /**
+     * Get a byte-array value from a given {@code HEXTORAW} argument.
+     *
+     * The provided {@code hexToRawValue} may be provided as the direct hex-string argument or the entire
+     * value may be wrapped with the {@code HEXTORAW} function call, which will be omitted.
+     *
+     * @param hexToRawValue the hex-to-raw string, never {@code null}
+     * @return byte array of decoded value, may be {@code null}
+     * @throws SQLException if there is a database exception
+     */
+    public byte[] getHexToRawByteArray(String hexToRawValue) throws SQLException {
+        final String data;
+        if (hexToRawValue.startsWith("HEXTORAW('") && hexToRawValue.endsWith("')")) {
+            data = hexToRawValue.substring(10, hexToRawValue.length() - 2);
+        }
+        else {
+            data = hexToRawValue;
+        }
+
+        return prepareQueryAndMap("select HEXTORAW(?) FROM DUAL", ps -> ps.setString(1, data), rs -> {
```

Comment:
Hey @Naros, sorry to hijack this already merged PR but I just stumbled over this code and noticed that it was recently committed, so I thought I would use this PR to comment.

This basically sends every LOB (or LOB chunk) again to the db server and reads back the converted value. So in the end each LOB gets transferred between database and connector 3 times. This is potentially a huge amount of data sent over the wire as we are dealing with LOBs here. Is there a reason why `oracle.sql.RAW#hexString2Bytes` was not used here?

Reply:
Captured this in https://issues.redhat.com/browse/DBZ-3556, so to keep track of this. Thanks, @TAregger!